x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/utils.ts

```diff
@@ -98,22 +108,46 @@ export const scrollAndBulkIndex = async (
     logger.warn('First bulk index was unsuccessful');
     return false;
   }
-  let newScrollId = someResult._scroll_id;
-  while (true) {
+
+  const totalHits =
+    typeof someResult.hits.total === 'number' ? someResult.hits.total : someResult.hits.total.value;
+  let size = someResult.hits.hits.length - 1;
+  logger.info(`first size: ${size}`);
+  let sortIds = someResult.hits.hits[0].sort;
+  if (sortIds == null) {
+    logger.warn('sortIds was empty on first search');
+    return false;
```

Comment:
I don't think this is an error condition right? Most signal rules will return empty as they didn't find anything as there is nothing to signal on and then when this runs it will find an empty data set which is totally normal.

It should only return an error if it is:

```ts
if (sortIds == null && totalHits > 0) {
  return false;
} else if (sortIds == null && totalHits == 0) {
  return true;
}
// else totalHits > zero and we have at least a first successful query
```

Reply:
This is a better idea thanks.